src/components/item/sharing/shareButton/CreateItemMembershipForm.tsx

```diff
@@ -134,59 +116,61 @@ const CreateItemMembershipForm = ({
     return returnedValue;
   };
 
-  const onChangeEmail: TextFieldProps['onChange'] = (event) => {
-    const newInvitation = {
-      ...invitation,
-      email: event.target.value,
-    };
-    setInvitation(newInvitation);
-    if (error) {
-      const isInvalid = checkForInvitationError(newInvitation);
-      setError(isInvalid);
-    }
-  };
-
   return (
     <Dialog onClose={handleClose} open={open}>
-      <DialogTitle>Share item</DialogTitle>
-      <DialogContent>
-        <Typography variant="body1">
-          {translateBuilder(BUILDER.SHARE_ITEM_FORM_INVITATION_TOOLTIP)}
-        </Typography>
-        <Stack
-          id={CREATE_MEMBERSHIP_FORM_ID}
-          direction="row"
-          alignItems="center"
-          justifyContent="center"
-          spacing={1}
-        >
-          <TextField
-            id={SHARE_ITEM_EMAIL_INPUT_ID}
-            variant="outlined"
-            label={translateBuilder(BUILDER.SHARE_ITEM_FORM_EMAIL_LABEL)}
-            helperText={error}
-            value={invitation.email}
-            onChange={onChangeEmail}
-            error={Boolean(error)}
-            sx={{ flexGrow: 1 }}
-          />
-          <ItemMembershipSelect
-            value={invitation.permission}
-            onChange={onChangePermission}
-            size="medium"
-          />
-        </Stack>
-      </DialogContent>
-      <DialogActions>
-        <Button variant="text">{translateCommon(COMMON.CANCEL_BUTTON)}</Button>
-        <Button
-          onClick={handleShare}
-          disabled={Boolean(error)}
-          id={SHARE_ITEM_SHARE_BUTTON_ID}
-        >
-          {translateBuilder(BUILDER.SHARE_ITEM_FORM_CONFIRM_BUTTON)}
-        </Button>
-      </DialogActions>
+      <DialogTitle>
+        {translateBuilder(BUILDER.SHARE_ITEM_FORM_TITLE, {
+          name: truncate(item.name, { length: ITEM_NAME_MAX_LENGTH }),
+        })}
+      </DialogTitle>
+      <Box component="form" onSubmit={handleSubmit(handleShare)}>
+        <DialogContent>
+          <Typography variant="body1">
+            {translateBuilder(BUILDER.SHARE_ITEM_FORM_INVITATION_TOOLTIP)}
+          </Typography>
+          <Stack
+            id={CREATE_MEMBERSHIP_FORM_ID}
+            direction="row"
+            alignItems="center"
+            justifyContent="center"
+            spacing={1}
+          >
+            <TextField
+              id={SHARE_ITEM_EMAIL_INPUT_ID}
+              variant="outlined"
+              label={translateBuilder(BUILDER.SHARE_ITEM_FORM_EMAIL_LABEL)}
+              helperText={errors.email?.message}
+              {...register('email', {
+                required: true,
+                validate: checkForInvitationError,
```

Comment:
You can pass multiple validation criteria:
```suggestion
                validate: { isEmail: (v) => validator.isEmail(v) || 'SOME_STRING', alreadyExists: checkForInvitationError},
```
It would allow to split up the validation function that you have currently.